datastore/tasks.js

```diff
@@ -175,78 +172,68 @@ function formatTasks (tasks) {
 }
 // [END format_results]
 
-if (module === require.main) {
-  var taskId;
-
-  switch (command) {
-    case 'new': {
-      addTask(input, function (err, taskKey) {
-        if (err) {
-          throw err;
-        }
-
-        taskId = taskKey.path.pop();
-
-        console.log('Task %d created successfully.', taskId);
-      });
-
-      break;
-    }
-    case 'done': {
-      taskId = parseInt(input, 10);
-
-      markDone(taskId, function (err) {
-        if (err) {
-          throw err;
-        }
-
-        console.log('Task %d updated successfully.', taskId);
-      });
+var cli = require('yargs');
+var makeHandler = require('../utils').makeHandler;
+
+var program = module.exports = {
+  addEntity: addTask,
+  updateEntity: markDone,
+  retrieveEntities: listTasks,
+  deleteEntity: deleteTask,
+  formatTasks: formatTasks,
+  main: function (args) {
+    // Run the command-line program
+    cli.help().strict().parse(args).argv;
+  }
+};
 
-      break;
-    }
-    case 'list': {
-      listTasks(function (err, tasks) {
-        if (err) {
-          throw err;
-        }
+cli
+  .demand(1)
+  .command('new <description>', 'Adds a task with a description <description>.', {}, function (options) {
+    addTask(options.description, function (err, taskKey) {
+      if (err) {
+        throw err;
+      }
 
-        console.log(formatTasks(tasks));
-      });
+      var taskId = taskKey.path.pop();
+      console.log('Task %d created successfully.', taskId);
+    });
+  })
+  .command('done <taskId>', 'Marks the specified task as done.', {}, function (options) {
+    markDone(options.taskId, function (err) {
+      if (err) {
+        throw err;
+      }
 
-      break;
-    }
-    case 'delete': {
-      taskId = parseInt(input, 10);
+      console.log('Task %d updated successfully.', options.taskId);
```

Comment:
Should this logic go in the parent function (e.g. `markDone`) as it did in the BigQuery samples, or in the `yargs` handler?